src/client/gpm_import_and_canon_name.c

```diff
@@ -2,6 +2,26 @@
 
 #include "gssapi_gpm.h"
 
+static int gpm_name_oid_to_static(gss_OID name_type, gss_OID *name_static)
+{
+#define ret_static(b) \
+    if (gpm_equal_oids(name_type, b)) { \
+        *name_static = b; \
+        return 0; \
+    }
+    ret_static(GSS_C_NT_USER_NAME);
+    ret_static(GSS_C_NT_MACHINE_UID_NAME);
+    ret_static(GSS_C_NT_STRING_UID_NAME);
+    ret_static(GSS_C_NT_HOSTBASED_SERVICE_X);
+    ret_static(GSS_C_NT_HOSTBASED_SERVICE);
+    ret_static(GSS_C_NT_ANONYMOUS);
+    ret_static(GSS_C_NT_EXPORT_NAME);
+    ret_static(GSS_C_NT_COMPOSITE_EXPORT);
+    ret_static(GSS_KRB5_NT_PRINCIPAL_NAME);
+    ret_static(gss_nt_krb5_name);
```

Comment:
Isn't this missing `GSS_KRB5_NT_ENTERPRISE_NAME`, `gss_nt_user_name`, `gss_nt_machine_uid_name`, `gss_nt_string_uid_name`, `gss_nt_service_name_v2`, `gss_nt_service_name`, `gss_nt_exported_name`?

It's probably safe to ignore iakerb and the old/wrong krb5 oids.

Reply:
This is really tricky, some of those are just that same thing, look at https://github.com/krb5/krb5/blob/master/src/lib/gssapi/generic/gssapi_generic.c#L182

After some discussion in #krbdev on IRC I decided to do exactly what is done here: https://github.com/krb5/krb5/blob/master/src/lib/gssapi/generic/oid_ops.c#L94

Plus the KRB5 names which I know can be returned by inspecting the krb5 code.
I am not aware of other mechanisms returning any other special names, should that be the case we'd have to add a global cache like for mechanims OIDs ...

Reply:
Got it, thanks for explaining that.